### DISCUSSION TITLE
Error executing gradle wrapper update command. It can be not a critical one though.

### DISCUSSION THREAD

**Discussion by eduardo-prospection** [renovatebot/renovate#29442] [Request Help] [answered]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

Bitbucket

### Please tell us more about your question or problem

It seems that the `gradle-wrapper` is trying to execute gradlew. 
I imagine it is doing that as a sort of validation.

Our gradle requires a few parameters to be passed to it. username/password for our internal maven repo.

                username = findProperty("nexusUser") as String
                password = findProperty("nexusPassword") as String

The error/log below fails on the line above.

I am unsure why Renovate tries to run Gradlew and how I would pass these parameters through.

I cannot reproduce it locally, it only happens when running on our CI/CD platform.

I don't have those parameters locally as env var. I unset them just to be sure

Renovate is trying to update gradle as

from distributionUrl=https\://services.gradle.org/distributions/gradle-8.3-bin.zip
to distributionUrl=https\://services.gradle.org/distributions/gradle-8.8-bin.zip

Kind regards

### Logs (if relevant)

<details><summary>Logs</summary>

```

`
> WARN: Error executing gradle wrapper update command. It can be not a critical one though. (repository=team/integration, branch=renovate/all-minor-patch)
>   --
>     | [2024-06-02T00:06:11Z]        "err": {
>     | [2024-06-02T00:06:11Z]          "cmd": "/bin/sh -c ./gradlew :wrapper --gradle-distribution-url https://services.gradle.org/distributions/gradle-8.8-bin.zip",
>     | [2024-06-02T00:06:11Z]          "stderr": "\nFAILURE: Build failed with an exception.\n\n* Where:\nBuild file '/tmp/renovate/repos/bitbucket/team/integration/build.gradle.kts' line: 34\n\n* What went wrong:\nnull cannot be cast to non-null type kotlin.String\n\n* Try:\n> Run with --stacktrace option to get the stack trace.\n> Run with --info or --debug option to get more log output.\n> Run with --scan to get full insights.\n> Get more help at https://help.gradle.org.\n\nBUILD FAILED in 1m 24s\n",

```

</details>

**Comment by rarkins** [comment, marked as answer]
Updating the gradle wrapper usually requires updating more than just a string in one file - the other wrapper files are usually checked into the repository.

Can you find a way that gradle can read environment variables? How do you normally provide such credentials in CI/CD?

**Reply by eduardo-prospection** [reply]
Tks, I realised the update was done by running gradle wrap.
I managed by naming the env var consistently through out
